python/lsst/daf/butler/core/storedFileInfo.py

```diff
@@ -19,87 +19,51 @@
-        a concrete composite.
-
-    See Also
-    --------
-    StorageInfo
+    This is an empty base class.  Datastore implementations are expected to
+    write their own subclasses.
     """
+    pass
 
-    __eq__ = slotValuesAreEqual
-    __slots__ = ("_formatter", "_path", "_storageClass", "_checksum", "_size")
-
-    def __init__(self, formatter, path, storageClass, checksum=None, size=None):
-        if isinstance(formatter, str):
-            # We trust that this string refers to a Formatter
-            pass
-        elif isinstance(formatter, Formatter) or (inspect.isclass(formatter) and
-                                                  issubclass(formatter, Formatter)):
-            formatter = formatter.name()
-        else:
-            raise ValueError(f"Supplied formatter '{formatter}' is not a Formatter")
-        self._formatter = formatter
-        assert isinstance(path, str)
-        self._path = path
-        assert isinstance(storageClass, StorageClass)
-        self._storageClass = storageClass
-        assert checksum is None or isinstance(checksum, str)
-        self._checksum = checksum
-        assert size is None or isinstance(size, int)
-        self._size = size
 
-    @property
-    def formatter(self):
-        """Full name of formatter (`str`).
-        """
-        return self._formatter
+@dataclass
+class StoredFileInfo(StoredDatastoreItemInfo):
+    """Datastore-private metadata associated with a file stored in a Datastore.
+    """
+    formatter: str
+    """Fully-qualified name of Formatter."""
 
-    @property
-    def path(self):
-        """Path to Dataset (`str`).
-        """
-        return self._path
+    path: str
+    """Path to dataset within Datastore."""
 
-    @property
-    def storageClass(self):
-        """StorageClass used (`StorageClass`).
-        """
-        return self._storageClass
+    storageClass: StorageClass
+    """StorageClass associated with Dataset."""
 
-    @property
-    def checksum(self):
-        """Checksum (`str`).
-        """
-        return self._checksum
+    checksum: Optional[str]
+    """Checksum of the serialized dataset."""
 
-    @property
-    def size(self):
-        """Size of stored object in bytes (`int`).
-        """
-        return self._size
+    file_size: int
+    """Size of the serialized dataset in bytes."""
 
-    def __repr__(self):
-        return f'{type(self).__qualname__}(path="{self.path}", formatter="{self.formatter}"' \
-            f' size={self.size}, checksum="{self.checksum}", storageClass="{self.storageClass.name}")'
+    def __post_init__(self):
+        # This modification prevents the data class from being frozen.
+        if isinstance(self.formatter, str):
+            # We trust that this string refers to a Formatter
+            pass
+        elif isinstance(self.formatter, Formatter) or \
+                (inspect.isclass(self.formatter) and issubclass(self.formatter, Formatter)):
```

Comment:
Why do we need the `inspect` check?  Does `issubclass` raise instead of returning `False` when given non-types?